rfc3164/rfc3164.go

```diff
@@ -169,16 +171,24 @@ func (p *Parser) parseTimestamp() (time.Time, error) {
     }
 
     sub = p.buff[p.cursor : tsFmtLen+p.cursor]
-    ts, err = time.ParseInLocation(tsFmt, string(sub), time.Local)
-    if err == nil {
-      /* Set Year on the Timestamp before converting to UTC so that time zone and
-         DST settings (which are year dependent) can be properly assessed in the
-         conversion. */
-      p.fixTimestampIfNeeded(&ts)
-
-      ts = ts.UTC()
-      found = true
-      break
+
+    // Only test the date pattern if the next character is a space (or the end of the message)
+    // This prevents us picking up the first 4 numbers of a hostname as a year
+    if p.cursor+tsFmtLen == p.l || p.buff[p.cursor + tsFmtLen] == ' ' {
+      ts, err = time.ParseInLocation(tsFmt, string(sub), time.Local)
+      if err == nil {
+        // Only use a pattern with a year for "reasonable" years 
+        if (ts.Year() > 0 && ts.Year() < 2000) || ts.Year() > 2100 {
```

Comment:
`year == 0` means the pattern had no year, and we need to guess based on the clock time. There is a risk we matched a four-digit hostname with all 0s. 

This could also be expressed by special-casing the two patterns with years (when we're iterating through the array of patterns, checking `i==0 || i == 3`).